### PR TITLE
Automatically determine correct AMI for VPN instance

### LINKED ISSUE #5: The VPN role shouldn't need default_ami to be set
AMI should be found and configured automatically

### PR DESCRIPTION
**Pull request by ejether**
Adds boto3 and methods to find ami-id based on default filters that are hard coded at present. updated options to shoose to configure or not configure the vpn env.yaml, added option to manually ovveride the vpn ami-id search
Added onwer id to the ami search

### CODE REVIEW

File: lib/pentagon/__init__.py [hunk 1]
```diff
@@ -59,6 +63,13 @@ class PentagonProject():
     _production_kubernetes_v_log_level = '<production_kubernetes_v_log_level>'
     _production_kubernetes_network_cidr = '<production_kubernetes_network_cidr>'
 
+    # VPN
+    _ami_owners = ['099720109477']  # Amazon AMI owner
+    _vpn_ami_id_placeholder = "<ami_id>"
+    _vpn_ami_filters = [{'Name': 'virtualization-type', 'Values': ['hvm']},
+                        {'Name': 'architecture', 'Values': ['x86_64']},
+                        {'Name': 'name', 'Values': ['ubuntu/images/hvm-ssd/ubuntu-xenial*']}]
```
Comment (philipcristiano) [inline lib/pentagon/__init__.py]: The previous AMI `ami-49c9295f` is a trusty AMI. Has the VPN role been tested with xenial?

File: setup.py [hunk 2]
```diff
@@ -42,7 +42,8 @@
         "pycrypto==2.6.1",
         "PyYAML==3.12",
         "shyaml==0.5.0",
-        "ansible==2.3.0.0"
+        "ansible==2.3.0.0",
+        "boto3==1.4.4"
```
Comment (philipcristiano) [inline setup.py:46]: Should this be added to `requirements.txt` as well?

File: lib/pentagon/__init__.py [hunk 3]
```diff
@@ -388,6 +403,44 @@ def __prepare_ansible_cfg_vars(self):
...
+                # Set EC2 environ vars for boto3 to use
+                os.environ['AWS_ACCESS_KEY'] = self._aws_access_key
+                os.environ['AWS_SECRET_KEY'] = self._aws_secret_key
+                os.environ['AWS_DEFAULT_REGION'] = self._aws_default_region
+                try:
+                    client = boto3.client('ec2')
+                    images = client.describe_images(Owners=self._ami_owners,Filters=self._vpn_ami_filters)
+                    self._vpn_ami_id = images['Images'][-1]['ImageId']
+                except Exception, e:
+                    logging.error("Encountered \" {} \" getting ami-id. VPN not configured fully".format(e))
```
Comment (philipcristiano) [inline lib/pentagon/__init__.py]: Please add more description here about what the user would need to do. It appears that `<ami_id>` will be dropped into the file.

**Comment by philipcristiano** [comment]
Can you add a more descriptive title for this PR in order to improve the Git history? Otherwise the merge commit would say:

```
Merge pull request #6

Fixes #5
```

which is more noise than signal if browsing the commit log.

File: lib/pentagon/__init__.py [hunk 4]
```diff
@@ -388,6 +403,44 @@ def __prepare_ansible_cfg_vars(self):
...
+
+                logging.info("Getting VPN ami-id from AWS")
+
+                # Set EC2 environ vars for boto3 to use
+                os.environ['AWS_ACCESS_KEY'] = self._aws_access_key
+                os.environ['AWS_SECRET_KEY'] = self._aws_secret_key
+                os.environ['AWS_DEFAULT_REGION'] = self._aws_default_region
+                try:
+                    client = boto3.client('ec2')
+                    images = client.describe_images(Owners=self._ami_owners,Filters=self._vpn_ami_filters)
```
Comment (philipcristiano) [inline lib/pentagon/__init__.py]: Pedantic but there appears to be a missing space after `self._ami_owners,`

File: setup.py (hunk 2 above)
Comment (ejether) [reply setup.py:46]: I don't think it needs to be. The setup.py and requirements.txt are separate concerns. setup.py adds the dependancies for the installation of the module. The requirements.txt seems to be more concerned with the build process. It is not leveraged when the module is `pip install`ed. Perhaps @jmound  can fill us in more on the use or this requirements.txt

File: lib/pentagon/__init__.py (hunk 3 above)
Comment (ejether) [reply lib/pentagon/__init__.py]: `<ami_id>` would be dropped into the file.
The output of the `pentagon start_project` script and the docs indicate what steps need to be taken. Do you think its correct to pollute the output log with steps to fix non-critical errors in every case?

Comment (philipcristiano) [reply lib/pentagon/__init__.py]: Maybe not non-critical, but it seems if the user explicitly tries to find an AMI, and Pentagon doesn't, it shouldn't create more work for the user. It looks like the `docs/vpn.md` has info for the AMI, maybe put a note here that the info can be found there?

Comment (ejether) [reply lib/pentagon/__init__.py]: Added a note to check the docs in case of warning

**Comment by philipcristiano** [comment]
LGTM

**Pull request merged**